d2/src/main/java/com/linkedin/d2/balancer/properties/UriPropertiesMerger.java

```diff
@@ -34,10 +39,16 @@ public UriProperties merge(String propertyName, Collection<UriProperties> proper
 
     String clusterName = propertyName;
     long maxVersion = -1;
+    long propertyVersion;
 
     for (UriProperties property : propertiesToMerge)
     {
-      maxVersion = Long.max(maxVersion, property.getVersion());
+      propertyVersion = property.getVersion();
+      maxVersion = Long.max(maxVersion, propertyVersion);
+      if (maxVersion == -1)
```

Comment:
Should we change the logic to 
```
if (propertyVersion <= -1) ?
```
As if the `propertiesToMerges` is (1, -1, -10), there is not warning.

Reply:
that's ok since the -1, -10 will warned already in [ZooKeeperEphemeralStore.java](https://github.com/linkedin/rest.li/pull/958/files#diff-77b01f0ad40699c361791ff221b474e25279aba3277103a4d7abe7ad5d1a2d3b). Here we just warn for the max version result.

Reply:
Make sense. Thanks